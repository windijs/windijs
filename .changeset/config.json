{
  "$schema": "https://unpkg.com/@changesets/config@2.1.1/schema.json",
  "changelog": "@changesets/cli/changelog",
  "commit": false,
  "fixed": [["@windijs/core", "windijs"]],
  "linked": [],
  "access": "public",
  "baseBranch": "main",
  "updateInternalDependencies": "patch",
  "ignore": []
}
