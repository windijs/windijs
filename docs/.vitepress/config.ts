import { defineConfig } from "vitepress";
import { mdRenderFilename } from "./markdown";

export default defineConfig({
  lang: "en-US",
  title: "Windi JS",
  cleanUrls: "without-subfolders",
  description: "The Utility-first CSS-in-JS Framework Powered by ES6 Proxy API and TypeScript.",
  markdown: {
    config(md) {
      md.use(mdRenderFilename);
    },
  },
  head: [
    ['link', { rel: "shortcut icon", type: "image/x-icon", href: "/windijs/favicon.ico" }]
  ],
  base: "/windijs/",
  themeConfig: {
    logo: "/logo.png",
    nav: [
      { text: "Guide", link: "/guide/what-is-windijs" },
      { text: "API", link: "/api" },
      { text: "Repl", link: "/repl" },
      { text: "Examples", link: "https://github.com/windijs/windijs/tree/main/examples" },
    ],
    sidebar: [
      {
        text: "Guide",
        collapsible: false,
        items: [
          {
            text: "What is Windi JS?",
            link: "/guide/what-is-windijs",
          },
          {
            text: "Getting Started",
            link: "/guide/getting-started",
          },
          {
            text: "Configuration",
            link: "/guide/configuration",
          },
          {
            text: "Optimization",
            link: "/guide/optimization",
          },
        ],
      },
      {
        text: "Customization",
        collapsible: false,
        items: [
          {
            text: "Theme",
            link: "/customization/theme",
          },
          {
            text: "Utility",
            link: "/customization/utility",
          },
          {
            text: "Variant",
            link: "/customization/variant",
          },
          {
            text: "Handler",
            link: "/customization/handler",
          },
          {
            text: "Loader",
            link: "/customization/loader",
          },
          {
            text: "Namer",
            link: "/customization/namer",
          },
          {
            text: "Plugins",
            link: "/customization/plugins",
          },
        ],
      },
      {
        text: "Integrations",
        collapsible: false,
        items: [
          {
            text: "API",
            link: "/integrations/api",
          },
          // {
          //   text: "CLI",
          //   link: "/integrations/cli",
          // },
          {
            text: "Vite",
            link: "/integrations/vite",
          },
          // {
          //   text: "Rollup",
          //   link: "/integrations/rollup",
          // },
          // {
          //   text: "Webpack",
          //   link: "/integrations/webpack",
          // },
        ],
      },
      {
        text: "Posts",
        collapsible: false,
        items: [
          {
            text: "How Windi JS Works",
            link: "/posts/how-windijs-works",
          },
          {
            text: "Ways of Using Windi JS",
            link: "/posts/ways-of-using-windijs",
          },
        ],
      },
    ],
    editLink: {
      pattern: "https://github.com/windijs/windijs/edit/main/docs/:path",
      text: "Edit this page on GitHub",
    },
    socialLinks: [
      { icon: "discord", link: "https://discord.gg/nJux6Gb52m" },
      { icon: "github", link: "https://github.com/windijs/windijs" },
    ],
    footer: {
      message: "Released under the MIT License.",
      copyright: "Copyright Â© 2022-present Raven Satir",
    },
  },
});
