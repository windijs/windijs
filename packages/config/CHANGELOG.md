# @windijs/config

## 1.0.6

### Patch Changes

- [`134427d6`](https://github.com/windijs/windijs/commit/134427d666a61704de77d52f0e6e4ffbfe085cd3) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: export package.json

* [`99ecf595`](https://github.com/windijs/windijs/commit/99ecf5956d7fa1124c1561d8f58b7a0a2f1af0f2) Thanks [@voorjaar](https://github.com/voorjaar)! - build: switch to es2018 for better tree shaking

* Updated dependencies [[`134427d6`](https://github.com/windijs/windijs/commit/134427d666a61704de77d52f0e6e4ffbfe085cd3), [`99ecf595`](https://github.com/windijs/windijs/commit/99ecf5956d7fa1124c1561d8f58b7a0a2f1af0f2)]:
  - @windijs/shared@1.0.5

## 1.0.5

### Patch Changes

- [`b38b5607`](https://github.com/windijs/windijs/commit/b38b5607dccfb2d5b22af15f1b62af9ad9caf3ae) Thanks [@voorjaar](https://github.com/voorjaar)! - feat: complete config define

* [`3bfacd6e`](https://github.com/windijs/windijs/commit/3bfacd6e745c01f650f1f5e7b65574585ee8c5bc) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: important config with string

* Updated dependencies [[`d285c569`](https://github.com/windijs/windijs/commit/d285c569ec7d4cdd0921f92c7dfe892e58c9cd75), [`831cae71`](https://github.com/windijs/windijs/commit/831cae71465e13a4d00188cca914710c54cd29cf)]:
  - @windijs/shared@1.0.4

## 1.0.5-beta.3

### Patch Changes

- [`3bfacd6e`](https://github.com/windijs/windijs/commit/3bfacd6e745c01f650f1f5e7b65574585ee8c5bc) Thanks [@voorjaar](https://github.com/voorjaar)! - fix: important config with string

## 1.0.5-beta.2

### Patch Changes

- [`b38b5607`](https://github.com/windijs/windijs/commit/b38b5607dccfb2d5b22af15f1b62af9ad9caf3ae) Thanks [@voorjaar](https://github.com/voorjaar)! - feat: complete config define

## 1.0.5-beta.1

### Patch Changes

- Updated dependencies [[`d285c569`](https://github.com/windijs/windijs/commit/d285c569ec7d4cdd0921f92c7dfe892e58c9cd75)]:
  - @windijs/shared@1.0.4-beta.1

## 1.0.5-beta.0

### Patch Changes

- Updated dependencies [[`831cae71`](https://github.com/windijs/windijs/commit/831cae71465e13a4d00188cca914710c54cd29cf)]:
  - @windijs/shared@1.0.4-beta.0

## 1.0.4

### Patch Changes

- af3c0c04: feat: add defineConfig interface
- Updated dependencies [939f0398]
  - @windijs/shared@1.0.3

## 1.0.3

### Patch Changes

- fix: should publish CHANGELOG.md
- Updated dependencies
  - @windijs/shared@1.0.2

## 1.0.2

### Patch Changes

- Updated dependencies
  - @windijs/shared@1.0.1

## 1.0.1

### Patch Changes

- 98ef7277: fix: package exports
