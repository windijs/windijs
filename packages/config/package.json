{
  "name": "@windijs/config",
  "version": "1.0.9",
  "description": "@windijs/config",
  "main": "dist/config.js",
  "module": "dist/config.mjs",
  "types": "dist/config.d.ts",
  "files": [
    "dist",
    "CHANGELOG.md"
  ],
  "keywords": [
    "css",
    "css-in-js",
    "windi",
    "windijs",
    "windicss"
  ],
  "exports": {
    "./package.json": "./package.json",
    ".": {
      "import": "./dist/config.mjs",
      "require": "./dist/config.js"
    }
  },
  "author": "Raven Satir",
  "license": "MIT",
  "typedoc": {
    "entryPoint": "./src/index.ts",
    "displayName": "@windijs/config"
  }
}
