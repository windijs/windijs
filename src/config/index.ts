export * from "./colors";
export * from "./bg";
export * from "./flex";
export * from "./grid";
export * from "./border";
export * from "./filter";
export * from "./opacity";
export * from "./sizing";
export * from "./spacing";
export * from "./positioning";
export * from "./interactivity";
export * from "./typography";
