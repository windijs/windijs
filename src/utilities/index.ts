export * from "./base";
export * from "./builder";
export * from "./style";
export * from "./api";
export * from "./bg";
export * from "./font";
export * from "./spacing";
