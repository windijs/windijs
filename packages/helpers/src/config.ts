import { Handler, StyleLoader, StyleObject } from "./types";

import { Theme } from "@windijs/config";

export type DarkModeConfig = "class" | "media" | false;

export type UtilitiesConfig = { [key: string]: Handler<unknown> | Array<Handler<unknown>> | StyleObject | UtilitiesConfig };

export type VariantsConfig = { [key: string]: string | string[] };

export interface Config<T extends object = Theme> {
  // presets?: Config[];
  // prefixer?: boolean;
  // attributify?: boolean | {
  //   prefix?: string
  //   separator?: string
  //   disable?: string[]
  // }
  separator?: string;
  important?: boolean | string;
  darkMode?: DarkModeConfig;
  theme?: T;
  // variantOrder?: string[];
  // plugins?: Plugin[];
  // handlers?: Handlers;
  // corePlugins?: (keyof BaseTheme)[] | string[] | { [ T in keyof BaseTheme ] : boolean } | { [ key:string ] : boolean };
  // prefix?: string;
  // exclude?: RegExp[];
  // alias?: { [key:string]: string };
  // shortcuts?: { [key:string]: Shortcut };
  // purge?: unknown;
  utilities?: UtilitiesConfig;
  variants?: VariantsConfig;
  styleLoader?: StyleLoader;
  [key:string]: any;
}

export function defineConfig (config: Config) {
  return config;
}
