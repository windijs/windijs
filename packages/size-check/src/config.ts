import { paddingConfig } from "@windijs/config";

export { paddingConfig };
