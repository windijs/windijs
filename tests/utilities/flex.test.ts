import { createUtility, flexDirectionUtility, flexGrowUtility, flexShrinkUtility, flexStretchUtility, flexUtility, flexWrapUtility, inlineFlexUtility } from "utilities";
import { flexDirectionConfig, flexGrowConfig, flexShrinkConfig, flexStretchConfig, flexWrapConfig } from "config";

test("Flex", () => {
  const flex = createUtility("flex").use(flexUtility()).init();
  expect(flex.css).toMatchSnapshot();
  expect(flex.meta).toMatchSnapshot();
});

test("Inline Flex", () => {
  const inlineFlex = createUtility("inlineFlex").use(inlineFlexUtility()).init();
  expect(inlineFlex.css).toMatchSnapshot();
  expect(inlineFlex.meta).toMatchSnapshot();

  const inline = createUtility("inline").case("flex", inlineFlexUtility()).init();
  // TODO: fix this ts error
  expect(inline.flex.css).toMatchSnapshot();

  const flex = createUtility("flex").case("inline", inlineFlexUtility()).init();
  expect(flex.inline.css).toMatchSnapshot();
});

test("Flex Direction", () => {
  const flex = createUtility("flex").use(flexDirectionUtility(flexDirectionConfig)).init();
  expect(flex.row.css).toMatchSnapshot();
  expect(flex.row.reverse.css).toMatchSnapshot();
  expect(flex.col.css).toMatchSnapshot();
  expect(flex.col.reverse.css).toMatchSnapshot();
});

test("Flex Wrap", () => {
  const flex = createUtility("flex").use(flexWrapUtility(flexWrapConfig)).init();
  expect(flex.wrap.css).toMatchSnapshot();
  expect(flex.wrap.reverse.css).toMatchSnapshot();
  expect(flex.nowrap.css).toMatchSnapshot();
});

test("Flex Stretch", () => {
  const flex = createUtility("flex").use(flexStretchUtility(flexStretchConfig)).init();
  expect(flex[1].css).toMatchSnapshot();
  expect(flex.auto.css).toMatchSnapshot();
  expect(flex.initial.css).toMatchSnapshot();
  expect(flex.none.css).toMatchSnapshot();
});

test("Flex Grow", () => {
  const flex = createUtility("flex").case("grow", flexGrowUtility(flexGrowConfig)).init();
  expect(flex.grow.css).toMatchSnapshot();
  expect(flex.grow[0].css).toMatchSnapshot();
});

test("Flex Shrink", () => {
  const flex = createUtility("flex").case("shrink", flexShrinkUtility(flexShrinkConfig)).init();
  expect(flex.shrink.css).toMatchSnapshot();
  expect(flex.shrink[0].css).toMatchSnapshot();
});

test("All Flex Utilities", () => {
  const flex = createUtility("flex")
    .use(flexUtility())
    .use(flexDirectionUtility(flexDirectionConfig))
    .use(flexWrapUtility(flexWrapConfig))
    .use(flexStretchUtility(flexStretchConfig))
    .case("grow", flexGrowUtility(flexGrowConfig))
    .case("shrink", flexShrinkUtility(flexShrinkConfig))
    .init();

  expect(flex.css).toMatchSnapshot();
  expect(flex.meta).toMatchSnapshot();
  expect(flex.row.css).toMatchSnapshot();
  expect(flex.col.reverse.css).toMatchSnapshot();
  expect(flex.wrap.reverse.css).toMatchSnapshot();
  expect(flex.initial.css).toMatchSnapshot();
  expect(flex.grow[0].css).toMatchSnapshot();
  expect(flex.shrink.css).toMatchSnapshot();
});
