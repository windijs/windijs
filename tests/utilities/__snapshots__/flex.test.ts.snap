// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All Flex Utilities 1`] = `
Object {
  "display": Array [
    "-webkit-box",
    "-ms-flexbox",
    "-webkit-flex",
    "flex",
  ],
}
`;

exports[`All Flex Utilities 2`] = `
Object {
  "props": Array [],
  "type": "css",
  "uid": "flex",
  "variants": Array [],
}
`;

exports[`All Flex Utilities 3`] = `
Object {
  "-ms-flex-direction": "row",
  "-webkit-box-direction": "normal",
  "-webkit-box-orient": "horizontal",
  "-webkit-flex-direction": "row",
  "flexDirection": "row",
}
`;

exports[`All Flex Utilities 4`] = `
Object {
  "-ms-flex-direction": "column-reverse",
  "-webkit-box-direction": "reverse",
  "-webkit-box-orient": "vertical",
  "-webkit-flex-direction": "column-reverse",
  "flexDirection": "column-reverse",
}
`;

exports[`All Flex Utilities 5`] = `
Object {
  "-ms-flex-wrap": "wrap-reverse",
  "-webkit-flex-wrap": "wrap-reverse",
  "flexWrap": "wrap-reverse",
}
`;

exports[`All Flex Utilities 6`] = `
Object {
  "-ms-flex": "0 1 auto",
  "-webkit-box-flex": "0",
  "-webkit-flex": "0 1 auto",
  "flex": "0 1 auto",
}
`;

exports[`All Flex Utilities 7`] = `
Object {
  "-ms-flex-positive": "0",
  "-webkit-box-flex": "0",
  "-webkit-flex-grow": "0",
  "flexGrow": "0",
}
`;

exports[`All Flex Utilities 8`] = `
Object {
  "-ms-flex-negative": "1",
  "-webkit-flex-shrink": "1",
  "flexShrink": "1",
}
`;

exports[`Flex 1`] = `
Object {
  "display": Array [
    "-webkit-box",
    "-ms-flexbox",
    "-webkit-flex",
    "flex",
  ],
}
`;

exports[`Flex 2`] = `
Object {
  "props": Array [],
  "type": "css",
  "uid": "flex",
  "variants": Array [],
}
`;

exports[`Flex Direction 1`] = `
Object {
  "-ms-flex-direction": "row",
  "-webkit-box-direction": "normal",
  "-webkit-box-orient": "horizontal",
  "-webkit-flex-direction": "row",
  "flexDirection": "row",
}
`;

exports[`Flex Direction 2`] = `
Object {
  "-ms-flex-direction": "row-reverse",
  "-webkit-box-direction": "reverse",
  "-webkit-box-orient": "horizontal",
  "-webkit-flex-direction": "row-reverse",
  "flexDirection": "row-reverse",
}
`;

exports[`Flex Direction 3`] = `
Object {
  "-ms-flex-direction": "column",
  "-webkit-box-direction": "normal",
  "-webkit-box-orient": "vertical",
  "-webkit-flex-direction": "column",
  "flexDirection": "column",
}
`;

exports[`Flex Direction 4`] = `
Object {
  "-ms-flex-direction": "column-reverse",
  "-webkit-box-direction": "reverse",
  "-webkit-box-orient": "vertical",
  "-webkit-flex-direction": "column-reverse",
  "flexDirection": "column-reverse",
}
`;

exports[`Flex Grow 1`] = `
Object {
  "-ms-flex-positive": "1",
  "-webkit-box-flex": "1",
  "-webkit-flex-grow": "1",
  "flexGrow": "1",
}
`;

exports[`Flex Grow 2`] = `
Object {
  "-ms-flex-positive": "0",
  "-webkit-box-flex": "0",
  "-webkit-flex-grow": "0",
  "flexGrow": "0",
}
`;

exports[`Flex Shrink 1`] = `
Object {
  "-ms-flex-negative": "1",
  "-webkit-flex-shrink": "1",
  "flexShrink": "1",
}
`;

exports[`Flex Shrink 2`] = `
Object {
  "-ms-flex-negative": "0",
  "-webkit-flex-shrink": "0",
  "flexShrink": "0",
}
`;

exports[`Flex Stretch 1`] = `
Object {
  "-ms-flex": "1 1 0%",
  "-webkit-box-flex": "1",
  "-webkit-flex": "1 1 0%",
  "flex": "1 1 0%",
}
`;

exports[`Flex Stretch 2`] = `
Object {
  "-ms-flex": "1 1 auto",
  "-webkit-box-flex": "1",
  "-webkit-flex": "1 1 auto",
  "flex": "1 1 auto",
}
`;

exports[`Flex Stretch 3`] = `
Object {
  "-ms-flex": "0 1 auto",
  "-webkit-box-flex": "0",
  "-webkit-flex": "0 1 auto",
  "flex": "0 1 auto",
}
`;

exports[`Flex Stretch 4`] = `
Object {
  "-ms-flex": "none",
  "-webkit-box-flex": "0",
  "-webkit-flex": "none",
  "flex": "none",
}
`;

exports[`Flex Wrap 1`] = `
Object {
  "-ms-flex-wrap": "wrap",
  "-webkit-flex-wrap": "wrap",
  "flexWrap": "wrap",
}
`;

exports[`Flex Wrap 2`] = `
Object {
  "-ms-flex-wrap": "wrap-reverse",
  "-webkit-flex-wrap": "wrap-reverse",
  "flexWrap": "wrap-reverse",
}
`;

exports[`Flex Wrap 3`] = `
Object {
  "-ms-flex-wrap": "nowrap",
  "-webkit-flex-wrap": "nowrap",
  "flexWrap": "nowrap",
}
`;

exports[`Inline Flex 1`] = `
Object {
  "display": Array [
    "-webkit-inline-box",
    "-ms-inline-flexbox",
    "-webkit-inline-flex",
    "inline-flex",
  ],
}
`;

exports[`Inline Flex 2`] = `
Object {
  "props": Array [],
  "type": "css",
  "uid": "inlineFlex",
  "variants": Array [],
}
`;

exports[`Inline Flex 3`] = `
Object {
  "display": Array [
    "-webkit-inline-box",
    "-ms-inline-flexbox",
    "-webkit-inline-flex",
    "inline-flex",
  ],
}
`;

exports[`Inline Flex 4`] = `
Object {
  "display": Array [
    "-webkit-inline-box",
    "-ms-inline-flexbox",
    "-webkit-inline-flex",
    "inline-flex",
  ],
}
`;
